File: plugins/RandomPlugin/src/plugins/quorum/Libraries/Compute/Random.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package plugins.quorum.Libraries.Compute;

/**
 * Java bytecode plugin for the "Random" class.
 * 
 * @author jeff
 */
public class Random {
    public java.lang.Object ___$$$Calling___$$$___Object$$$___ = null;
    private java.util.Random random = new java.util.Random();
    
    public void InitializeNative() {
        long seed = System.currentTimeMillis();
        random.setSeed(seed);
    }
    
    public int RandomIntegerNative(int maximum) {
        return random.nextInt(maximum + 1);
    }
    
    public void SetSeed(double seed) {
        random.setSeed((long)seed);
    }
    
    public int RandomInteger() {
        return random.nextInt(2147483647);
    }
    
    public double RandomNumber() {
        return random.nextDouble();
    }
    
    public boolean RandomBoolean() {
        return random.nextBoolean();
    }

}

File: Quorum3/Library/Standard/Plugins/javascript/Libraries/Compute/Random.js
function plugins_quorum_Libraries_Compute_Random_() {
    this.InitializeNative = function () {};
    
    this.RandomIntegerNative$quorum_integer = function (maximum) {
        return Math.floor(Math.random() * (maximum + 1));
    };
    
    this.SetSeed$quorum_number = function (seed) {};
    
    this.RandomInteger = function () {
        return Math.floor(Math.random() * (Number.MAX_VALUE + 1));
    };
    
    this.RandomNumber = function () {
        return Math.random();
    };

    this.RandomBoolean = function () {
        if (Math.random() < 0.5)
            return true;
        else
            return false;
    };

}